docs/content/getting-started/data-out/analyze-and-log.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Analyze the data and log the results
+order: 3
+---
+
+
+
+In the previous sections, we explored our data and exported it to a Pandas dataframe. In this section, we will analyze the data to extract a "jaw open state" signal and log it back to the viewer.
+
+
+
+## Analyze the data
+
+This is admittedly not the most complicated part of this guide. We already identified that thresholding the `jawOpen` signal at 0.15 is all we need to produce a binary "jaw open state" signal.
+
+In the [previous section](export-dataframe.md#inspect-the-dataframe), we prepared a flat, floating point column with the signal of interest called `"jawOpen"`. Let's add a boolean column to our Pandas dataframe to hold our jaw open state:
+
+```python
+df["jawOpenState"] = df["jawOpen"] > 0.15
+```
+
+
+## Log the result back to the viewer
+
+The first step to log the data is to initialize the logging SDK to direct it to the exact same recording that we just analyzed. For this, both the application ID and the recording ID must match. Here is how it is done:
+
+```python
+rr.init(
+    recording.application_id()
+    recording_id=recording.recording_id(),
+)
+rr.connect()
+```
+
+_Note_: When automating data analysis, it is typically preferable to log the results to an distinct RRD file next to the source RRD (using `rr.save()`). In such case, it is also valid to use the same app ID and recording ID. This allows opening both the source and result RRDs in the viewer, which will display both data under the same recording.
+
+We will log our jaw open state data in two forms:
+1. As a standalone `Scalar` component, to hold the raw data.
+2. As a `Text` component on the existing bounding box entity, such that we obtain a textual representation of the state in the visualization.
+
+Here is how to log the data as a scalar:
+
+```python
+rr.send_columns(
+    "/jaw_open_state",
+    times=[rr.TimeSequenceColumn("frame_nr", df["frame_nr"])],
+    components=[
+        rr.components.ScalarBatch(df["jawOpenState"]),
+    ],
+)
+```
+
+With use the [`rr.send_column()`](../../howto/send_columns.md) API to efficiently send the entire column of data in a single batch.
+
+Next, let's log the same data as `Text` component:
+
+```python
+target_entity = "/video/detector/faces/0/bbox"
+rr.log_components(target_entity, [rr.components.ShowLabels(True)], static=True)
+rr.send_columns(
+    target_entity,
+    times=[rr.TimeSequenceColumn("frame_nr", df["frame_nr"])],
+    components=[
+        rr.components.TextBatch(np.where(df["jawOpenState"], "OPEN", "CLOSE")),
+    ],
+)
+```
+
+Here we first log the [`ShowLabel`](../../reference/types/components/show_labels.md) component as static to enable the display of the label. Then, we use `rr.send_column()` again to send an entire batch of text labels. We use the [`np.where()`](https://numpy.org/doc/stable/reference/generated/numpy.where.html) to produce a label matching the state for each timestamp.
```

Comment:
this one failed for me with 
```
D:\dev\rerun-io\rerun\test.py:34: RerunWarning: send_columns: ValueError(All times and components in a batch must have the same length. Expected length: 355 but got: 1 for component: rerun.components.Text)
```
looks like the np array insn't interpreted as it should be or np.where does not what we think it does. Will check

Reply:
Huh. That seems to be what https://github.com/rerun-io/rerun/pull/7689 was supposed to fix!

Reply:
yeah, I didn't hit this.

Reply:
my build was outdated, that was why!